File: Java/DSA/Algorithms/Sorting/BubbleSort.java
// Bubble Sort Algorithm

import java.util.Scanner;

public class BubbleSort{
    public static void main(String[] args) {
        
        int size;

        Scanner in = new Scanner(System.in);

        System.out.print("Enter the number of the inputs: ");

        size = in.nextInt();

        // array to hold the inputs
        int[] arr = new int[size];

        for (int i=0; i<size; i++)
        {
            arr[i] = in.nextInt();
        }

        in.close();

        bubbleSort(arr);
        System.out.println("Sorted Array is: ");
        for (int i=0; i<size; i++)
        {
            System.out.print(arr[i]+" ");
        }

    }

    public static void bubbleSort(int[] arr)
    {
        int size = arr.length;

        for(int i=0; i<size-1; i++)
        {
            boolean swapped = false;
            for (int j = 0; j<size-1-i; j++)
            {
                if (arr[j] > arr[j+1])
                {
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;

                    swapped = true;
                }
            }

            if(swapped == false)
            {
                // numbers are already sorted;
                break;
            }
        }
    }
}

File: C++/Sorting Algorithms/bubbleSort.cpp

// Bubble sort algorithm..

#include <iostream>
using namespace std;

void bubble_sort(int arr[], int size)
{
    for ( int i=0; i<size-1; i++)
    {
        for (int j=0; j<size-1-i; j++)
        {
            if (arr[j]>arr[j+1])
            {
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
}

int main()
{

    int arr[4] = {16, 19, 11, 15};

    // sorting the above array using bubble sort..
    // call the function bubble_sort()
    bubble_sort(arr, 4);
    cout << "Sorted array is: " << endl;

    for (int i=0; i<4; i++)
    {
        cout << arr[i] << " ";
    }


    return 0;
}